scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -1705,6 +1705,100 @@ def visit_functiondef(self, node):
+            "enum-must-inherit-case-insensitive-enum-meta",
+            "Inherit CaseInsensitiveEnumMeta.",
+        ),
+    }
+    options = (
+        (
+            "ignore-enum-must-be-uppercase",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not be capitalized.",
+            },
+        ),
+        (
+            "ignore-enum-must-inherit-case-insensitive-enum-meta",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not inherit CaseInsensitiveEnumMeta.",
+            },
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(CheckEnum, self).__init__(linter)
+
+    def visit_classdef(self, node):
+        """Visits every enum class.
+
+        :param node: ast.ClassDef
+        :return: None
+        """
+        try:
+            enum_class = False
+            case_insensitive_meta = False
+            
+            #Python3 format declares CaseInsensitiveEnumMeta as a metaclass in enum classes
+            if node.declared_metaclass():
+                if node.declared_metaclass().name:
+                    if node.declared_metaclass().name == "CaseInsensitiveEnumMeta":   # or node.declared_metaclass().name == "_CaseInsensitiveEnumMeta":
+                        case_insensitive_meta = True
+                        enum_class = True
+            if not case_insensitive_meta:
+                # Python2 format uses CaseInsensitiveEnumMeta as a base of the classDef node
+                for base in node.bases:
+                    if isinstance(base, astroid.Call):
+                        for arg in base.args:
+                            if arg.name:
+                                if arg.name == "CaseInsensitiveEnumMeta":
+                                    case_insensitive_meta = True
+                                    enum_class = True
+                                    break
+                    elif base.name == "Enum":
+                        enum_class = True
+               
+                    
+            # If it is an enum class, make sure all enums in the class are capitalized
+            if enum_class:
+                for nod in node.body:
+                    if isinstance(nod, astroid.Assign):
+                        # An Enum is an assign statement
+                        for x in nod.targets[0].name:
+                            if x.islower():
+                                # if the name has any lowercase letters
+                                self.add_message(
+                                        "enum-must-be-uppercase", node=nod.targets[0], confidence=None
+                                    )
+                                break
+                if not case_insensitive_meta:
+                    self.add_message(
```

Comment:
How does it look if an enum class has both of the pylint errors?

Reply:
I will add a test!